src/script/standard.cpp

```diff
@@ -317,8 +320,30 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, vector<vector<unsi
                     // Check that the number is encoded with the minimum possible number of bytes. This is also different 
                     // before the fork but this way is consistent with interpreter
                     static const bool REQ_MINIMAL = true;
-                    const int32_t nHeight = CScriptNum(vchBlockHeight, REQ_MINIMAL, sizeof(int32_t)).getint();
- 
+                    int32_t nHeight = -1;
+                    try
+                    {
+                        nHeight = CScriptNum(vchBlockHeight, REQ_MINIMAL, sizeof(int32_t)).getint();
+                    }
+                    catch(const scriptnum_error& e)
+                    {
+                        LogPrintf("%s: %s():%d - OP_CHECKBLOCKATHEIGHT nHeight 0x%s not minimally encoded (err=%s)\n",
+                            __FILE__, __func__, __LINE__, HexStr(vchBlockHeight.begin(), vchBlockHeight.end()), e.what());
+                        break;
+                    }
+                    catch(...)
+                    {
+                        LogPrint("%s: %s():%d - unexpected exception\n", __FILE__, __func__, __LINE__);
+                        break;
+                    }
+
+                    if (!CheckMinimalPush(vchBlockHeight, hopcode))
```

Comment:
EvalScript seems to be checking minimal push only if opcode <= OP_PUSHDATA4
https://github.com/HorizenOfficial/zen/blob/2aaf4b3bedb6eedd6e8a7a45dbc15b95bf7172b0/src/script/interpreter.cpp#L298

Reply:
The opcode value is red in a different context here compared to the other parser, where the same logic is used only when we are checking specifically the RP height and using the associated opcode, stored during a previous parser step.